joomla/joomla-cms
# [4.0] Removing repeatable field plugin

Pull Request for Issue #23659.

### Summary of Changes
We have the subform field and don't need the repeatable field anymore. This removes that field as requested in #23659 and previously done in https://github.com/continga/joomla-cms/commit/71f831d2a78fdc5340ef5d60352cdb7fa23e04ec.

This PR removes the repeatable field and adds a method to the update script which converts the existing repeatable fields to the subform field type. For all repeatable fields it inserts new fields and links those to the subfields field. To not display them twice in the edit view, the category for the new fields is always set to "none". Please notice that a field in the repeatable field of type "number" can NOT be converted, since we don't have a field of type number. You can however change this later on to integer for example.

### Testing Instructions
1. Setup at least one repeatable field with at least one field.
2. Copy /plugins/fields/repeatable to a temporary folder.
3. Apply this PR and then copy back the above folder again.
4. Call the uninstallRepeatableFieldsPlugin() method from /administrator/components/com_admin/script.php. This can be done for example by using this code:
	```php
	public function update()
	{
		require_once JPATH_ADMINISTRATOR . '/components/com_admin/script.php';
		\JoomlaInstallerScript::uninstallRepeatableFieldsPlugin();
	}
	```
	Add this to for example /administrator/components/com_content/src/Controller/ArticleController.php in line 53 and then call /administrator/index.php?option=com_content&task=article.update Before you do that, you need to modify the the access modifier of that method to public. So edit /administrator/components/com_admin/script.php line 235 from protected to public.
5. Go to the fields view of the component where you have the repeatable field and see that it has been converted to subfields.
